utils/startup/installdir.cpp

```diff
@@ -71,6 +78,62 @@ const string StartUp::installDir()
     return *fInstallDirp;
 }
 
+/* static */
+mutex StartUp::fTmpDirLock;
+/* static */
+string* StartUp::fTmpDirp = 0;
+
+/* static */
+const string StartUp::tmpDir()
+{
+    mutex::scoped_lock lk(fTmpDirLock);
+
+    if (fTmpDirp)
+        return *fTmpDirp;
+
+#ifdef _MSC_VER
+    fTmpDirp = new string("C:\\Calpont\Tmp");
+    string cfStr = IDBreadRegistry("");
+
+    if (!cfStr.empty())
+        *fTmpDirp = cfStr;
+
+#else
+
+	//check for non-root user 
+    const char* p = getenv("HOME");
+	string homedir = p;
+
+    if (homedir == "/root")
+	{
+		Config* sysConfig = Config::makeConfig();
+
+		string TempFileDir;
+
+		try
+		{
+			TempFileDir = sysConfig->getConfig("SystemConfig", "TempFileDir");
+		}
+		catch (...)
+		{}
+
+		fTmpDirp = new string("/tmp");
+		
+		*fTmpDirp = *fTmpDirp + TempFileDir;
+	}
+	else
+	{
+		// non-root user
+		fTmpDirp = new string(homedir);
+
+		*fTmpDirp = *fTmpDirp + "/.tmp";
```

Comment:
remind me why we're hiding the tmp dir?  But also shouldn't we just return what's in the config file?